crates/iota-types/src/iota_system_state/mod.rs

```diff
@@ -190,6 +192,7 @@ pub trait IotaSystemStateTrait {
 #[enum_dispatch(IotaSystemStateTrait)]
 pub enum IotaSystemState {
     V1(IotaSystemStateV1),
+    V2(IotaSystemStateV2),
```

Comment:
The second system state version also needs to be added in the [advance_epoch_safe_mode](https://github.com/iotaledger/iota/blob/0d6616b33284cad5638687f156aaad30aa1d6138/crates/iota-types/src/iota_system_state/mod.rs#L85) function.